From 8d17ad214da42d485bc8dc540b3cc42f0b751587 Mon Sep 17 00:00:00 2001
From: Hans-Christoph Steiner <hans@eds.org>
Date: Wed, 26 Oct 2011 22:14:45 -0400
Subject: [PATCH 1/2] make it build on a plain Mac OS X/Fink system

---
 Makefile     |    6 +++---
 fux_kinect.h |    2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index e45d658..8b4a718 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 # change to your local directories!
 PD_APP_DIR = /Applications/Pd-extended.app/Contents/Resources
-PD_DIR = /Users/xcorex/Documents/Documents/Projects/Puredata/PdSource/Pd-0.42.5-extended/pd
-GEM_DIR = /Users/xcorex/Documents/Documents/Projects/Puredata/PdSource/Pd-0.42.5-extended/Gem
+PD_DIR = $(PD_APP_DIR)/include
+GEM_DIR = $(PD_APP_DIR)/include
 # build flags
 
 INCLUDES = -I$(PD_DIR)/include
@@ -30,7 +30,7 @@ SOURCES = fux_kinect
 
 all:
 	g++ $(LDFLAGS) $(INCLUDES) $(CPPFLAGS) -o $(SOURCES).o -c $(SOURCES).cpp
-	g++ -o $(SOURCES).$(EXTENSION) -undefined dynamic_lookup -arch i386 -dynamiclib -mmacosx-version-min=10.3 -undefined dynamic_lookup -framework QuickTime -framework Carbon -framework AGL -framework OpenGL -arch i386 ./*.o -L/sw/lib -lfreeimage -lstdc++ -ldl -lz -lm -lpthread -lfreenect -L$(PD_DIR)/bin -L$(PD_DIR)
+	g++ -o $(SOURCES).$(EXTENSION) -undefined dynamic_lookup -arch i386 -dynamiclib -mmacosx-version-min=10.3 -undefined dynamic_lookup -framework QuickTime -framework Carbon -framework AGL -framework OpenGL -arch i386 ./*.o -L/sw/lib -lstdc++ -ldl -lz -lm -lpthread -lfreenect -L$(PD_DIR)/bin -L$(PD_DIR)
 	rm -fr ./*.o
 deploy:
 	rm -fr $(PD_APP_DIR)/extra/$(SOURCES).$(EXTENSION)
diff --git a/fux_kinect.h b/fux_kinect.h
index 4b53e21..1329f58 100644
--- a/fux_kinect.h
+++ b/fux_kinect.h
@@ -17,7 +17,7 @@ LOG
 
 #include "Base/GemBase.h"
 #include "Base/GemPixUtil.h"
-#include "FreeImage.h"
+//#include "FreeImage.h"
 #include "libfreenect.h"
 
 #include <pthread.h>
-- 
1.7.6.4

