--- pybuffer.cpp	2008-09-25 22:01:58.000000000 +0200
+++ pybuffer-working.cpp	2008-09-25 22:01:12.000000000 +0200
@@ -21,31 +21,20 @@
 
 #ifdef PY_ARRAYS
 
-#ifdef PY_NUMARRAY
-#	if FLEXT_OS == FLEXT_OS_MAC
-#		include <Python/numarray/libnumarray.h>
-#	else
-#		include <numarray/libnumarray.h>
-#	endif
-
-static NumarrayType numtype = tAny;
-inline bool arrsupport() { return numtype != tAny; }
-
-#else
-#	if defined(PY_NUMPY)
-#		include <numpy/arrayobject.h>
+#if defined(PY_NUMERIC)
+#		include <Numeric/arrayobject.h>
 #	else
 #		if FLEXT_OS == FLEXT_OS_MAC
-#			include <Python/numarray/arrayobject.h>
+#			include <Python/Numeric/arrayobject.h>
 #		else
-#			include <numarray/arrayobject.h>
+#			include <Numeric/arrayobject.h>
 #		endif
 #	endif
 
 	static PyArray_TYPES numtype = PyArray_NOTYPE;
 	inline bool arrsupport() { return numtype != PyArray_NOTYPE; }
 #endif
-#endif
+
 
 
 PyObject *pybase::py_arraysupport(PyObject *self,PyObject *args)
