--- vexp_if.c	2004-03-04 19:12:51.000000000 +0100
+++ vexp_if.c-old	2004-03-04 19:10:48.000000000 +0100
@@ -888,7 +888,7 @@
  *  the result pointer 
  */
 int
-max_ex_tab(struct expr *expression,fts_symbol_t s,struct ex_ex *arg,struct ex_ex *optr)
+max_ex_tab(struct expr *exp,fts_symbol_t s,struct ex_ex *arg,struct ex_ex *optr)
 {
 #ifdef PD
 	t_garray *garray;
@@ -900,7 +900,7 @@
 	{
 		optr->ex_type = ET_FLT;
 		optr->ex_flt = 0;
-		pd_error(expression, "no such table '%s'", s->s_name);
+		pd_error(exp, "no such table '%s'", s->s_name);
 		return (1);
 	}
 	optr->ex_type = ET_FLT;
@@ -915,7 +915,7 @@
 		break;
 
 	default:	/* do something with strings */
-		pd_error(expression, "expr: bad argument for table '%s'\n", fts_symbol_name(s));
+		pd_error(exp, "expr: bad argument for table '%s'\n", fts_symbol_name(s));
 		indx = 0;
 	}
 	if (indx < 0) indx = 0;
@@ -933,13 +933,13 @@
 }
 
 int
-max_ex_var(struct expr *expression, fts_symbol_t var, struct ex_ex *optr)
+max_ex_var(struct expr *exp, fts_symbol_t var, struct ex_ex *optr)
 {
 	optr->ex_type = ET_FLT;
 	if (value_getfloat(var, &(optr->ex_flt))) {
 		optr->ex_type = ET_FLT;
 		optr->ex_flt = 0;
-		pd_error(expression, "no such var '%s'", var->s_name);
+		pd_error(exp, "no such var '%s'", var->s_name);
 		return (1);
 	}
 	return (0);
